### PR TITLE
updates from version v0.8

### LINKED ISSUE #12: old software version provided
Please update the released software.

### PR DESCRIPTION
**Pull request by kreier**
no more negative voltages
offset for solar2 corrected
credentials in separate file

### CODE REVIEW

File: software/SolarAISVN/SolarAISVN.ino [hunk 1]
```diff
@@ -1,27 +1,30 @@
...
-// 2020/06/15
+// Solar- and windmeter at AISVN v0.8 
+// 2020/06/16
 //
-//
-// voltages converted on ESP32
-// submission time back to 120 seconds
+// no more negative voltages
+// offset for solar2 corrected
+// credentials in separate file
```
Comment (ssis-robotics) [inline software/SolarAISVN/SolarAISVN.ino:6]: Isn't that the solution to issue #12 ?

File: software/SolarAISVN/SolarAISVN.ino [hunk 2]
```diff
@@ -161,14 +164,18 @@ void measureVoltages() {
   //    32,      33,       34,       35,   14,   26,   27,     12,   13
   // solar, battery, currentA, currentB, load, wind, dump, solar2, LiPo  
   voltage[0] = int((4096 - voltage[0]) * 7.52 - 1000);  // pin32 solar    voltage divider 10k : 1.2 k Ohm 1:1
+  if(voltage[0] < 0) voltage[0] = 0;
```
Comment (ssis-robotics) [inline software/SolarAISVN/SolarAISVN.ino:167]: Negative voltage would be adjusted to zero.

File: software/SolarAISVN/SolarAISVN.ino [hunk 3]
```diff
@@ -161,14 +164,18 @@ void measureVoltages() {
...
   voltage[1] = int((4096 - voltage[1]) * 7.52 - 1000);  // pin33 battery  voltage divider 10k : 1.2 k Ohm 1:1
   //voltage[2] = int((voltage[2]) * 0.804 + 129);         // pin34 voltage solar minus green LED
   //voltage[3] = int((voltage[3]) * 0.804 + 129);         // pin35 voltage solar minus green LED minus 0.1 Ohm serial
   voltage[2] = int((voltage[3] - voltage[2]) * 6.75);   // voltage difference pin35 - pin34 x 8.4 is corrent (x0.804)
   voltage[4] = int((4096 - voltage[4]) * 7.52 - 1000);  // pin14 load     voltage divider 10k : 1.2 k Ohm 1:1
+  if(voltage[4] < 0) voltage[4] = 0;  
   voltage[5] = int((4096 - voltage[5]) * 7.52 - 1000);  // pin26 wind     voltage divider 10k : 1.2 k Ohm 1:1
-  voltage[6] = int((voltage[6]) * 1.608 + 258);         // pin27 dump     not connected yet
+  if(voltage[5] < 0) voltage[5] = 0;  
+  voltage[6] = int((voltage[6]) * 0.804 + 129);         // pin27 temp     not connected yet
```
Comment (ssis-robotics) [inline software/SolarAISVN/SolarAISVN.ino:176]: Included some specific calculation to have no negative voltages measured

**Review by ghost — COMMENTED** [review]
I think issue #12 is solved hereby.

File: software/SolarAISVN/SolarAISVN.ino [hunk 4]
```diff
@@ -1,27 +1,30 @@
-// Solar- and windmeter at AISVN v0.7 
-// 2020/06/15
+// Solar- and windmeter at AISVN v0.8 
+// 2020/06/16
 //
-//
-// voltages converted on ESP32
-// submission time back to 120 seconds
```
Comment (aisvn-data) [inline software/SolarAISVN/SolarAISVN.ino:6]: This improves the evaluation of the data later.

File: software/SolarAISVN/SolarAISVN.ino (hunk 2 above)
Comment (aisvn-data) [reply software/SolarAISVN/SolarAISVN.ino:167]: Good update!

File: software/SolarAISVN/SolarAISVN.ino (hunk 3 above)
Comment (aisvn-data) [reply software/SolarAISVN/SolarAISVN.ino:176]: Professional solution.

**Review by aisvn-data — APPROVED** [review]

**Pull request merged**